test/arrow.jl

```diff
@@ -0,0 +1,38 @@
+@testset "Arrow support" begin
+    @testset "Interval (concrete)" begin
+        col = [Interval{Closed,Unbounded}(1, nothing)]
+
+        table = (; col)
+        t = Arrow.Table(Arrow.tobuffer(table))
+
+        @test eltype(t.col) == Interval{Int, Closed, Unbounded}
+        @test t.col == col
+    end
+
+    @testset "Interval (non-concrete)" begin
+        col = [
+            Interval{Closed, Closed}(1, 2),
+            Interval{Closed, Open}(2, 3),
+            Interval{Unbounded, Open}(nothing, 4),
+        ]
+
+        table = (; col)
+        t = Arrow.Table(Arrow.tobuffer(table))
+
+        @test eltype(t.col) == Interval{Int}
+        @test t.col == col
+    end
+
+    @testset "AnchoredInterval" begin
+        zdt_start = ZonedDateTime(2016, 8, 11, 1, tz"America/Winnipeg")
+        zdt_end = ZonedDateTime(2016, 8, 12, 0, tz"America/Winnipeg")
+        col = HE.(zdt_start:Hour(1):zdt_end)
+
+        table = (; col)
+        t = Arrow.Table(Arrow.tobuffer(table))
+
+        # Arrow.jl converts all Period types into Second
+        @test_broken eltype(t.col) == HourEnding{ZonedDateTime, Open, Closed}
```

Comment:
I'll probably end up disabling `AnchoredInterval` support to start with because of this

Reply:
We can't explicitly override this behaviour?

Reply:
I've forgotten a bunch about this PR but it should be possible to encode the period span in the serialized Arrow form. I do remember I was mainly focused on `Interval` support so I think this comment was more about that the implementation currently in place is incomplete.